### PR TITLE
De-dupe details and main bootstrap page + rolling update considerations

### LINKED ISSUE #362: Rolling updates in K8s with bootstrap
Test / document how best to use, including CoordinatedShutdown/SBR.

### LINKED ISSUE #388: Deduplicate Bootstrap details page
The bootstrap details page has two redundant explanations. It also overlaps with the "How It Works" section from the overview page.

### PR DESCRIPTION
**Pull request by chbatey**
Fixes #388 
Fixes #362

### CODE REVIEW

**Comment by dwijnand** [comment]
Fixes #388?

File: docs/src/main/paradox/bootstrap/index.md [hunk 1]
```diff
@@ -86,6 +86,71 @@ is strictly defined and is as follows:
...
+Akka Cluster can handle hard failures using a downing provider such as Lightbend's split brain resolver discussed below.
+However this should not be relied upon for regular rolling redeploys. Features such as `ClusterSinglton`s and `ClusterSharding`
+can safely restart actors on new nodes far quicker when it is certain that a node has shutdown rather than crashed. 
+
+Graceful leaving will happen with the default settings as it is part of Coordinated Shutdown. Just ensure that
+a node is sent a `SIGTERM` and not a `SIGKILL`. Environments such as Kubernetes will do this, it is important to ensure 
+that if JVM is wrapped with a script that it forwards the signal. Upon receiving a `SIGTERM` Coordinated shutdown will:
+
+* Perform a `Cluster(system).leave` on its self
+* The `Leader` will then move the node to `Exiting` and then `Removed` allowing any shards to be shutdown gracefully
```
Comment (patriknw) [inline docs/src/main/paradox/bootstrap/index.md]: I don't think we need to bother the user with introducing the concept of leader here.

The status of the member will be changed to `Exiting` while allowing any shards to be shutdown gracefully and `ClusterSingleton`s to be migrated if this was the oldest node. Finally the node is removed from the Akka Cluster membership.

File: docs/src/main/paradox/bootstrap/index.md [hunk 2]
```diff
@@ -86,6 +86,71 @@ is strictly defined and is as follows:
...
+
+Nodes can crash causing cluster members to become unreachable. This is a tricky problem as it is not
+possible to distinguish between a network partition and a node failure. To rectify this in an automated manner,
+Lightbend provides [Split Brain Resolver](https://developer.lightbend.com/docs/akka-commercial-addons/current/split-brain-resolver.html)
+as a feature of the Lightbend Subscription. This module has a number of strategies that can ensure that the cluster
+continues to function during network partitions and node failures.
+
 ## Bootstrap Recipes
 
 To see how to configure and use bootstrap in various environments such as Kubernetes, see @ref[recipes](recipes.md).
```
Comment (patriknw) [inline docs/src/main/paradox/bootstrap/index.md]: recipes.md is not in any toc and therefore the left side of that page looks strange

File: docs/src/main/paradox/bootstrap/details.md [hunk 3]
```diff
@@ -1,112 +1,40 @@
-# Bootstrap details
+# Initial bootstrap
```
Comment (patriknw) [inline docs/src/main/paradox/bootstrap/details.md]: Why change that to Initial? It describes the full process in more detail.

File: docs/src/main/paradox/bootstrap/details.md [hunk 4]
```diff
@@ -123,39 +51,19 @@ even if the DNS system is not completely consistent.
...
-akka.cluster.shutdown-after-unsuccessful-join-seed-nodes = 40s
-```
-
-Additionally, nodes can crash causing cluster members to become unreachable. This is a tricky problem as it is not
-possible to distinguish between a network partition and a node failure. To rectify this in an automated manner,
-Lightbend provides [Split Brain Resolver](https://developer.lightbend.com/docs/akka-commercial-addons/current/split-brain-resolver.html)
-as a feature of the Lightbend Subscription. This module has a number of strategies that can ensure that the cluster
-continues to function during network partitions and node failures.
-
 ## Customising Join Behavior
```
Comment (patriknw) [inline docs/src/main/paradox/bootstrap/details.md]: I think we try to use American spelling: Customizing

File: docs/src/main/paradox/bootstrap/details.md [hunk 5]
```diff
@@ -123,39 +51,19 @@ even if the DNS system is not completely consistent.
 If however we are talking about an inconsistent DNS lookup response during the Initial Bootstrap, the nodes will be delayed
 forming the cluster as they expect the lookups to be consistent, this is checked by the stable-margin configuration option.
 
-For complete safety of the Initial Bootstrap it is recommended to set the `akka.management.cluster.bootstrap.contact-point-discovery.required-contact-point-nr`
+For complete safety of the Initial Bootstrap it is recommended to set the `contact-point-discovery.required-contact-point-nr`
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/details.md]: I moved this section to the main page under "initial deployment" keeping this page to just be the detailed version of the process

File: docs/src/main/paradox/bootstrap/index.md (hunk 2 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/index.md]: Fixed

File: docs/src/main/paradox/bootstrap/details.md [hunk 6]
```diff
@@ -1,112 +1,40 @@
...
+    - Other nodes notice that A has started returning *itself* as a seed-node in the Contact Point responses.
     - Any node, including X, immediately joins such seed node that it has observed in the Contact Point process.
     - Nodes continue probing the other nodes, and eventually will notice any of the existing nodes that are part of the cluster,
-      and immediately join it. This phase is referred to as "epidemic joining".
+      and immediately join it. 
     - Eventually all nodes have joined the same cluster, the process is complete.
 
 The illustration below may be of help in visualising this process:
 
-![project structure](../images/bootstrap-forming-cluster.png)
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/details.md]: I'll leave it in for now, not sure it helps

File: docs/src/main/paradox/bootstrap/details.md (hunk 3 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/details.md]: Reverting

File: docs/src/main/paradox/bootstrap/index.md [hunk 7]
```diff
@@ -86,6 +86,71 @@ is strictly defined and is as follows:
...
+
+Akka Cluster can handle hard failures using a downing provider such as Lightbend's split brain resolver discussed below.
+However this should not be relied upon for regular rolling redeploys. Features such as `ClusterSinglton`s and `ClusterSharding`
+can safely restart actors on new nodes far quicker when it is certain that a node has shutdown rather than crashed. 
+
+Graceful leaving will happen with the default settings as it is part of Coordinated Shutdown. Just ensure that
+a node is sent a `SIGTERM` and not a `SIGKILL`. Environments such as Kubernetes will do this, it is important to ensure 
+that if JVM is wrapped with a script that it forwards the signal. Upon receiving a `SIGTERM` Coordinated shutdown will:
+
+* Perform a `Cluster(system).leave` on its self
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/index.md]: :+1:

File: docs/src/main/paradox/bootstrap/details.md (hunk 4 above)
Comment (chbatey) [reply docs/src/main/paradox/bootstrap/details.md]: Okay :)

**Comment by chbatey** [comment]
Pushed review feedback

**Pull request merged**